CONTRIBUTING.md

```diff
@@ -0,0 +1,107 @@
+# Contributing to RKE #
+
+Thanks for taking the time to contribute to RKE!
+
+Of course, contributing code is more than welcome! To keep things simple, if you're fixing a small issue, you can simply submit a PR and we will pick it up. However, if you're planning to submit a bigger PR to implement a new feature or fix a relatively complex bug, please continue reading.
+
+### Opening PRs and organizing commits
+PRs should generally address only 1 issue at a time. If you need to fix two bugs, open two separate PRs. This will keep the scope of your pull requests smaller and allow them to be reviewed and merged more quickly.
+
+Each PR needs to be linked to a Github issue in [rancher project](https://github.com/rancher/rancher) which provides as much detail as possible on the commit at its motivation. If it is a bug fix, please describe how to reproduce it in the issue. Make the team or Project Manager aware of the Github issue so that they can set the correct milestone. Milestones are very important for release and QA tracking.
+
+Generally, pull requests should consist of a single logical commit. However, if your PR is for a large feature, you may need a more logical breakdown of commits. This is fine as long as each commit is a single logical unit. Changes generated by `go mod tidy` or `go generate` should go in separate commits.
+
+Pull requests should target the release/vX.Y branch instead of master. The latest code in RKE is included in such release branches.
+
+### Writing good commit messages
+Git commit messages should explain the how and why of your change and be separated into a brief subject line followed by a more detailed body. When in doubt, follow this guide for good commit messages and you can’t go wrong: https://chris.beams.io/posts/git-commit/.
```

Comment:
I don't think we generally following this approach and using PR description in lieu of "detailed body". Our commit messages are generally short - for better or worse.

Reply:
ok, I can change that. I don't care where things are described as long as they are described :)